### ISSUE TITLE
Accuracy Doubting

### ISSUE THREAD

**Issue by AdaLovelemon** [gszfwsb/NCFM#10]
Dear authors,

Thank you for open-sourcing this admirable project — it’s been a great learning opportunity to reproduce the experiments.

After some debugging (especially on the Sampling Net — we’ve tried our best to keep it faithful to your implementation as described in the paper), the code runs smoothly. However, we’ve encountered a discrepancy in the reported results. Specifically, when setting IPC = 1 on CIFAR-10, we’re consistently getting ~36% accuracy, rather than the ~50% reported in the paper.

Could you kindly share the distilled image .pt files you used, or provide more detailed hyperparameter settings/configurations that might help us reproduce the reported results more faithfully?

Thank you again for your valuable work and for taking the time to look into this.

Best regards,

![Image](https://github.com/user-attachments/assets/4dff8da0-8bfa-40d4-96fc-675f46bf9f9b)

**Comment by AdaLovelemon** [comment]
After weaks of hard work, we cannot reproduce your experiment results with your provided code.

I cannot get your results, please give me your hyperparameters' configurations, and another important thing is that, please open-source condensed images (pts), please 

Evaluation model weights on IPC10-cifar10 is really helpful for our followings 

Thank you for your insightful paper

**Comment by Namelessness5** [comment]
You may check pt files right here:https://huggingface.co/maomaocun/NCFM. However, I am also not able to get the reported accuracy.

**Comment by maomaocun** [comment]
Thank you for your patience. We initially released a version of the sampling network using the AdamW optimizer. We have now updated it to use the SGD optimizer with a learning rate of 1e-3 and the same momentum as the image optimizer.

**Comment by haiduo** [comment]
> You may check pt files right here:https://huggingface.co/maomaocun/NCFM. However, I am also not able to get the reported accuracy.

+1, Using the sampling net seems to be worse than not using it; it seems like the sampling net is not useful.
